packages/next/src/server/lib/router-utils/block-cross-site.ts

```diff
@@ -0,0 +1,65 @@
+import type { Duplex } from 'stream'
+import type { IncomingMessage, ServerResponse } from 'webpack-dev-server'
+import { parseUrl } from '../../../lib/url'
+import net from 'net'
+
+export const blockCrossSite = (
+  req: IncomingMessage,
+  res: ServerResponse | Duplex,
+  allowedOrigins: string[],
+  activePort: string
+): {
+  finished?: boolean
```

Comment:
should this just return a boolean rather than an object? then perhaps we can just call it `shouldBlockCrossSite` and then it's a bit more idiomatic when we use it eg:

```ts
if (shouldBlockCrossSite(req, res, allowedOrigins, `${opts.port}`)) {
  return
}
```

Reply:
Sounds alright to me, was just matching our routing pattern of checking for `.finished`